themes/default/theme/stencil/src/components/ai/ai.tsx

```diff
@@ -264,10 +292,47 @@ export class PulumiAI {
         if (this.storeUnsubscribe) {
             this.storeUnsubscribe();
         }
+
+        if (this.pingListener) {
+            clearInterval(this.pingListener);
+        }
     }
 
-    private onConnected(_event: Event) {
+    private onConnected(data: CreateConnectionResponse) {
         this.connectionStatus = "Connected";
+        this.conversationId = data.conversationId;
+        
+        if (this.existingConversationId) {
+            console.log("exisiting conversation", this.existingConversationId);
+            this.client.getConversation(this.existingConversationId);
+        }
+
+        // We ping the backend every 30 seconds to ensure
+        // we are keeping connections alive.
+        this.pingListener = setInterval(() => this.client.ping(), 30000);
+    }
+
+    private onGetConverstation(data: GetConversationResponse) {
+        const versions: Version[] = data.conversation.map(c => {
+            return {
+                prompt: c.prompt,
+                source: c.response,
+                language: c.language,
+                markup: this.addButtons(marked.marked.parse(c.response)),
+            };
+        });
+
+        this.currentVersion = versions[versions.length - 1];
+        this.versions = Array.from(versions);
+        this.currentVersion = null;
+
+        this.running = false;
+        this.scroll();
+        this.clear();
+        this.focus();
+
+        // Wait a bit for the versions to render, then colorize them.
+        setTimeout(() => this.highlight(this.outputVersions), 250);
```

Comment:
Might be nice to eliminate some of this duplication, but we could also come back to it.